test/testlib/tls.go

```diff
@@ -167,6 +192,32 @@ func RotateTLSCertificates(t *testing.T, options *helm.Options, namespaceName st
 	AwaitDatabaseUp(t, namespaceName, admin0, "demo", 2)
 }
 
+func AwaitAdminKeystoreReload(t *testing.T, namespace string, podName string, adminPem string, timeout time.Duration) {
+	options := k8s.NewKubectlOptions("", "", namespace)
+	adminPem = strings.TrimSpace(adminPem)
+	Await(t, func() bool {
+		data, err := k8s.RunKubectlAndGetOutputE(t, options, "exec", podName, "-c", "admin", "--",
+			"nuocmd", "--show-json", "get", "certificate-info")
+		require.NoError(t, err)
+		err, certificateInfo := UnmarshalCertificateInfo(data)
+		require.NoError(t, err)
+		// Wait for all APs to reload their keystore files and return the
+		// updated server certificate
+		var notUpdated []string
+		for admin, info := range certificateInfo.ServerCertificates {
+			actual := strings.TrimSpace(info.CertificatePem)
+			if actual != adminPem {
+				notUpdated = append(notUpdated, admin)
+			}
+		}
+		if len(notUpdated) > 0 {
+			t.Logf("Server certificates not updated for APs %s", notUpdated)
+			return false
+		}
+		return true
+	}, timeout)
```

Comment:
So ConfigMaps/Secrets do get reloaded? I have not been able to find documentation that confirms this and was always under the impression that this was not the case. Is this a newer feature?

Reply:
This is done by _kubelet_ and I don't think is a new feature. The secrets mounted with `subPath` doesn't receive updates though. This was how TLS secrets were mounted before which was unfortunate.
For more information on the Secret updates, see [here](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-files-from-a-pod).

Reply:
Thanks for the pointer. It's good that ConfigMaps/Secrets mounted as files are reloaded. I thought we were going to have to rely on K8s API access to fetch the data.